Everywhere: Fix badly-formatted and unused includes

BenWiederhake opened the pull request:
This cleans up our includes:
- by fixing badly-formatted includes (e.g. `#include "AK/StdLibExtras.h"`)
- by removing completely unused includes.

This re-uses the least-controversial changes from #16764 and #15282, as well as a lot of new instances that have been introduced in the meantime.

trflynn89 commented on Userland/Libraries/LibJS/Runtime/CanonicalIndex.h:
```diff
@@ -6,7 +6,6 @@
 
 #pragma once
 
-#include <AK/Concepts.h>
```
This is used.

BenWiederhake replied:
Oh right, `FloatingPoint`! You're right, I'll remove that. It compiles without any issue though.

trflynn89 commented on Userland/Applications/Browser/History/HistoryModel.h:
```diff
@@ -6,7 +6,7 @@
 #pragma once
 
-#include "Applications/Browser/History.h"
+#include "../History.h"
```
:eyes: IMO relative include paths like this are worse than just writing `<Applications/Browser/History.h>`, but I don't really have a stronger argument than "yuck".

BenWiederhake replied:
Yeah, that particular line is awful. I'll go with the system-include syntax like you suggested.

BenWiederhake commented:
Changes since last push:
- Reverted change to LibJS/Runtime/CanonicalIndex.h (thanks @trflynn89!)
- Slightly nicer include in Browser/History/HistoryModel.h (thanks @trflynn89!)
- Rebased for convenience
